modules/compliance-tests/src/smithy4s/compliancetests/internals/eq/EqSchemaVisitor.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.compliancetests.internals.eq
+
+import smithy4s.compliancetests.internals.eq.Smithy4sEqInstances._
+import cats.implicits.catsSyntaxEq
+import cats.kernel.Eq
+import smithy4s._
+import smithy4s.schema.{Schema, _}
+
+import java.util.UUID
+import smithy4s.capability.EncoderK
+
+object EqSchemaVisitor extends SchemaVisitor[Eq] { self =>
```

Comment:
as a user, it would be nice to have it as public. I know, API surface...

Reply:
End users don't want that, it's wildly inefficient. I however think we should tweak the dynamic model to respect structural equality.

Reply:
@kubukoz Can I lend a hand here ? do you know what has to be done?